contracts/BC_fusion/GovToken.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20BurnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20PermitUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "./System.sol";
+
+interface IStakePool {
+    function getPooledBNB(address account) external view returns(uint256);
+}
+
+contract GovToken is
+    System,
+    Initializable,
+    ERC20Upgradeable,
+    ERC20BurnableUpgradeable,
+    ERC20PermitUpgradeable,
+    ERC20VotesUpgradeable
+{
+    string private constant NAME = "BSC Governance Token";
+    string private constant SYMBOL = "govBNB";
+
+    // validator StakePool contract => user => amount
+    mapping(address => mapping(address => uint256)) public mintedMap;
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize()
+    public
+    initializer
+    {
+        __ERC20_init(NAME, SYMBOL);
+        __ERC20Burnable_init();
+        __ERC20Permit_init(NAME);
+        __ERC20Votes_init();
+    }
+
+    function sync(address validatorPool, address account) external onlyStakeHub {
+        uint256 latestBNBAmount = IStakePool(validatorPool).getPooledBNB(account);
```

Comment:
why not make it as a parameter of this function so that we dont have to do cross contract invoke.

Reply:
Because `StakeHub` also requires a cross contract call to get the latest pooled BNB from `StakePool` contract